src/components/timeline/TrackContextMenu.js

```diff
@@ -516,13 +534,46 @@ class TimelineTrackContextMenu extends PureComponent<Props> {
     );
   }
 
+  renderHideAllTracksByType() {
+    const { rightClickedTrack, globalTracks, localTracksByPid } = this.props;
+    if (rightClickedTrack === null) {
+      return null;
+    }
+    const trackIndex = rightClickedTrack.trackIndex;
+    if (rightClickedTrack.type === 'global') {
+      return (
+        <MenuItem
+          key={trackIndex + 'hide-all'}
+          preventClose={false}
+          data={(rightClickedTrack, globalTracks, localTracksByPid)}
+          onClick={this._toggleGlobalTrackVisibility}
+        >
+          Hide all {`"${this.getRightClickedTrackType(rightClickedTrack)}"`}{' '}
+          tracks
+        </MenuItem>
+      );
+    }
+    return (
+      <MenuItem
+        key={trackIndex + 'hide-all'}
+        preventClose={false}
+        data={(rightClickedTrack, globalTracks, localTracksByPid)}
+        onClick={this._toggleGlobalTrackVisibility}
```

Comment:
onClick events are still using the same function as `Hide "xyz"` item. Can you please add your own click even that hides all the tracks by their type like I mentioned on the previous review. Please let me know if you have any question.

Reply:
### Hello @canova I had added a function but a bit confused about my method:
```
getRightClickedTrackType(rightClickedThreadIndex: TrackReference): string {
const { localTracksByPid, globalTracks } = this.props;
if (rightClickedThreadIndex.type === 'global') {
  return localTracksByPid[rightClickedThreadIndex.trackIndex];
}
const globalTrackNames = globalTracks.get(
  rightClickedThreadIndex.trackIndex
);
if (globalTrackNames === undefined) {
  console.error(
    'Expected to find a local and globaltrack name for the given pid.'
  );
  return 'Unknown Track';
}
 return globalTracks[rightClickedThreadIndex.trackIndex];
}
```